src/api/spec/support/vcr.rb

```diff
@@ -3,7 +3,7 @@
 VCR.configure do |config|
   config.cassette_library_dir = 'spec/cassettes'
   config.hook_into :webmock
-  config.default_cassette_options = { record: :new_episodes }
+  config.default_cassette_options = { record: :new_episodes, re_record_interval: 1.days }
```

Comment:
In general I like the idea :+1:
We just need to keep in mind that this will create a lot of "noise"

Reply:
Why would you want to commit the noise?

Reply:
the faker makes it more noiser than it needs to be - is there a way to seed it? :)

Reply:
But leave the cassettes completly out of git and purely for developers to speed up their tests? Sounds even better

Reply:
1. detecting backend regressions affecting frontend
2. detecting testsuite bugs hidden by cassettes
3. actually be able to test the backend in rspec tests

(not necessarly in that order)

Reply:
1. happens only theoretically
2. is something that should be fixed
3. we have a backend test suite

I'm really not sure what you are proposing here is worth doing to cover 2.

Reply:
we have a backend test suite - possibly. But if I want to test in rspec tests that webui changes actually end up in backend, I do need one running. And this does not belong in a backend test suite.

Reply:
I see actually very little *benefit* of mocking the backend - taking that frontend and backend are 2 sides of the story. Especially if the mocking is frozen in git.

Reply:
> I see actually very little benefit of mocking the backend

I understand that. But we see a benefit, speed of the CI cycle and easier setup.

Reply:
I don't see the ease of setup either. See https://github.com/openSUSE/open-build-service/pull/5142 - you get crappy undebuggable problems with VCRs.

How much the CI cycle is sped up is to be seen, measuring it atm.

Reply:
What is undebuggable in #5142?

Reply:
> I don't see the ease of setup either.

[Erm...](https://github.com/openSUSE/open-build-service/blame/master/src/api/script/start_test_backend)

Reply:
Well, this was born out of the need to have *something* - and is unrelated to mocking the backend. Not sure this script would have even have existed when containers existed 2010. But you can hardly take one wrong as argument against another wrong (especially if there are 8 years in between)

Reply:
I'm not really interested in how the script does the things, it's legacy code. But **what** this script does is what **needs to be done**, right?

Now if you want to do integration tests without mocking you also have to do the nasty parts of that script. Cover timing, cleanup, injection of fixtures, separation from the dev environment etc. Hence you will complicate the CI cycle setup. This is what we have decided against.

Why don't you roll with VCR, fix the problems you see, instead of trying to replace it at the first bump in the road before you really have any experience with it?

Reply:
What I'm trying to do here is making sure the tests run fine with and without VCRs. Backend VCRs should be an optimization not a requirement

Reply:
> making sure the tests run fine with and without VCRs

Yes please do that. It's rather easy to spot the errors and fix them.

But the cost of introducing a backend into the CI cycle is just to high.

Reply:
I actually doubt that, but we can talk about it once we can verify :)